### ISSUE TITLE
Fix mapping queueing 

### ISSUE THREAD

**Issue by elizabethmcd** [Arcadia-Science/hifi2genome#21]
### Description of the bug

Currently minimap2 queuing matches random indexes to reads, which by chance sometimes works, sometimes doesn't. Have to explicitly pair by the meta map of the correct assembly to corresponding reads. 

Found this issue with command run below when from the report found a couple samples with 0% mapping, the wrong assembly is paired with incorrect reads. 

### Command used and terminal output

```console
minimap2 \
    -x map-hifi -a \
    -t 6 \
    bacillus_cereus_971.assembly.fasta.mmi \
    m64004_210929_143746.bc2001.fq.gz | \
    samtools view -@ 6 -bS | \
    samtools sort -@ 6 -o acineobacter_baumanii_AYE.sorted.bam
samtools index acineobacter_baumanii_AYE.sorted.bam acineobacter_baumanii_AYE.bam.bai
```

**Comment by elizabethmcd** [comment]
Something along the lines of: 
```
ch_index
        .cross(ch_fastq) { it -> it[-1] }
        .flatten()
        .collate(13)
        .map { it -> [ it[7], it[8], it[0], it[1], it[2], it[3], it[4], it[5] ] } // [ sample, fastq, fasta, sizes, gtf, bed, is_transcripts, index ]
        .set { ch_index }
```

I don't understand what the `collate()` does though

**Comment by elizabethmcd** [comment]
fixed in #23
